plugins/shodan/shodan.go

```diff
@@ -0,0 +1,25 @@
+package shodan
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/needsauth"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+)
+
+func ShodanCLI() schema.Executable {
+	return schema.Executable{
+		Name:    "Shodan Command-Line Interface",
+		Runs:    []string{"shodan"},
+		DocsURL: sdk.URL("https://cli.shodan.io"),
+		NeedsAuth: needsauth.IfAll(
+			needsauth.NotForHelpOrVersion(),
+			needsauth.NotWithoutArgs(),
```

Comment:
shall we also exclude `shodan init` from here?

Reply:
I appended `needsauth.NotForExactArgs("init")` to the arguments of `needsauth.IfAll()`.